apps/ejabberd/src/ejabberd_binary.erl

```diff
@@ -28,3 +29,16 @@ string_to_binary(S) when is_list(S) ->
     end;
 string_to_binary(B) when is_binary(B) ->
     B.
+
+
+join([], _) ->
```

Comment:
Ah, true, I think I saw it too. I wonder if it still holds with current OTP versions (AFAIR those were done with R16).